addons/mail/models/res_users.py

```diff
@@ -103,6 +108,13 @@ def write(self, vals):
             self.env['mail.channel'].search([('group_ids', 'in', sel_groups)])._subscribe_users()
         return write_res
 
+    def unlink(self):
+        # Un-subscribe deleted users from non public channels
+        self.mapped('partner_id.channel_ids').filtered(lambda c: c.public != 'public').write({
```

Comment:
Could be defined in a single method IMO + write a comment about the reason why we only kick partners from non-public channels. 

I had to look at the task history to understand why. So I won't be the only one who doesn't understand why. 

Thanks.